### PR TITLE
EN 968 groups page

### PR DESCRIPTION
**Pull request by mcarta-rt**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: src/api/groups.js [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+import { GROUPS_OK_PAGE_1, GROUPS_OK_PAGE_2 } from 'test/mocks/groups';
+import throttle from 'util/throttle';
 
-// eslint-disable-next-line
-export const getApiGroups = () => (
-  new Promise((resolve) => {
-    resolve(GROUPS);
-  })
-);
+export const getGroups = (page, params) => new Promise((resolve) => {
```
Comment (alessioscalici) [inline src/api/groups.js]: Needs to be aligned with the new way we manage API and API tests (see https://github.com/entando/designtime-admin-app/pull/54)

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: src/state/groups/actions.js [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
   },
 });
 
 // thunk
-export const fetchGroups = () => dispatch =>
-  getApiGroups().then((data) => {
-    dispatch(addGroups(data.payload));
+export const fetchGroups = (page = { page: 1, pageSize: 10 }, params = '') => dispatch =>
+  new Promise((resolve) => {
+    getGroups(page, params).then((response) => {
```
Comment (alessioscalici) [inline src/state/groups/actions.js]: What if the promise get rejected? The Promise would remain pending forever.
You need to add `.catch()` blocks in internal promises, something like

```javascript
export const fetchGroups = (page = { page: 1, pageSize: 10 }, params = '') => dispatch =>
  new Promise((resolve, reject) => {
    getGroups(page, params).then((response) => {
      if (response.ok) {
        response.json().then((data) => {
          dispatch(setGroups(data.payload));
          dispatch(setPage(data.metaData));
          resolve();
        })
        .catch(reject);
      } else {
        resolve();
      }
    })
    .catch(reject);
  });
```

**Pull request merged**